src/AutoMapper/Mappers/DynamicMappers.cs

```diff
@@ -21,6 +21,7 @@ public class FromDynamicMapper : IObjectMapExpression
                 destination = (TDestination) (!context.ConfigurationProvider.AllowNullDestinationValues
```

Comment:
I did refactor some of these in some of the other mappers where this code was.  Things like Nullable you know the type can only be nullable, so you can just new it up as such.  Maybe these you can do that to some extent, but it wasn't as obvious as the others were.